sdk/python/feast/infra/passthrough_provider.py

```diff
@@ -47,14 +52,17 @@ def update_infra(
         partial: bool,
     ):
         set_usage_attribute("provider", self.__class__.__name__)
-        self.online_store.update(
-            config=self.repo_config,
-            tables_to_delete=tables_to_delete,
-            tables_to_keep=tables_to_keep,
-            entities_to_keep=entities_to_keep,
-            entities_to_delete=entities_to_delete,
-            partial=partial,
-        )
+
+        # Call update only if there is an online store
+        if self.online_store:
+            self.online_store.update(
+                config=self.repo_config,
+                tables_to_delete=tables_to_delete,
+                tables_to_keep=tables_to_keep,
+                entities_to_keep=entities_to_keep,
+                entities_to_delete=entities_to_delete,
+                partial=partial,
+            )
 
     def teardown_infra(
         self, project: str, tables: Sequence[FeatureView], entities: Sequence[Entity],
```

Comment:
don't we need to add the same `if self.online_store` check for `teardown_infra`?

if we add this check, we should be able to keep the type annotation in `get_online_store_from_config` as Oleksii suggested

Reply:
I changed some methods, function signatures and returns. Could you please review? I made those changes based on the output of `make lint-python`

Reply:
Could I get one more review @felixwang9817